### ISSUE TITLE
[BUG] Wrong normal distribution formula

### ISSUE THREAD

**Issue by mirkoklukas** [genjax-community/genjax#474]
Unless I am missing somehting the formula in 
https://github.com/probcomp/genjax/blob/main/src/genjax/_src/generative_functions/distributions/scipy/normal.py
should be: `- jnp.log(std) - 1/2*jnp.log(2*jnp.pi) - 1/2*z**2`

```python
n = 100
x   = np.random.randn(n)
mu  = np.random.randn(n)
std = np.random.rand(n)

z = (x - mu) / std

reference       = jax.scipy.stats.norm.logpdf(x, mu, std)
currently_wrong = genjax.normal.logpdf(x, mu, std)
corrected       = - jnp.log(std) - 1/2*jnp.log(2*jnp.pi) - 1/2*z**2


err_curr = jnp.abs(reference - currently_wrong)
err_new  = jnp.abs(reference - corrected)

assert jnp.mean(err_curr) > jnp.mean(err_new)
```

**Comment by femtomc** [comment]
Closed on `nightly` @ c827133c497918f508e5e2222547efcea13d307d. Will keep open until hits `main`.

**Comment by sritchie** [comment]
Closing now, in prep for tomorrow's release.
